sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/immunizations/EditImmunizationSteps.java

```diff
@@ -0,0 +1,79 @@
+package org.sormas.e2etests.steps.web.application.immunizations;
+
+import static org.sormas.e2etests.pages.application.cases.EditCasePage.*;
+import static org.sormas.e2etests.pages.application.immunizations.CreateNewImmunizationPage.MEANS_OF_IMMUNIZATIONS_COMBOBOX;
+import static org.sormas.e2etests.pages.application.immunizations.EditImmunizationPage.DISEASE_COMBOBOX;
+import static org.sormas.e2etests.pages.application.immunizations.EditImmunizationPage.FACILITY_COMBOBOX_IMMUNIZATION_INPUT;
+import static org.sormas.e2etests.pages.application.users.EditUserPage.FACILITY_CATEGORY_COMBOBOX_INPUT;
+import static org.sormas.e2etests.pages.application.users.EditUserPage.FACILITY_TYPE_COMBOBOX_INPUT;
+
+import com.google.common.truth.Truth;
+import cucumber.api.java8.En;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import javax.inject.Inject;
+import javax.inject.Named;
+import lombok.SneakyThrows;
+import org.sormas.e2etests.helpers.WebDriverHelpers;
+import org.sormas.e2etests.pojo.web.Immunization;
+import org.sormas.e2etests.services.ImmunizationService;
+
+public class EditImmunizationSteps implements En {
+
+  public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("M/d/yyyy");
+  public static Immunization aImmunization;
+  private final WebDriverHelpers webDriverHelpers;
+
+  @SneakyThrows
+  @Inject
+  public EditImmunizationSteps(
+      WebDriverHelpers webDriverHelpers,
+      ImmunizationService immunizationService,
+      @Named("ENVIRONMENT_URL") String environmentUrl) {
+    this.webDriverHelpers = webDriverHelpers;
+
+    When(
+        "I check the created data is correctly displayed on Edit immunization page",
+        () -> {
+          aImmunization = collectImmunizationData();
+          Truth.assertThat(aImmunization.getDisease())
```

Comment:
if the POJO created first would be the same with the collected POJO, use just Truth.assertThat(POJO1).equals(POJO2).
If not, use soft asserts, and add also messages (they lack in the tests, and in case of failure, it's hard to see exactly the reason in allure report). Also, by using soft assert we can go through all asserts and catch multiple failures